libtock/led.c

```diff
@@ -1,17 +1,37 @@
 #include "led.h"
 
 int led_count(void) {
-  return command(DRIVER_NUM_LEDS, 0, 0, 0);
+  syscall_return_t ret = command2(DRIVER_NUM_LEDS, 0, 0, 0);
+  if (ret.type == TOCK_SYSCALL_SUCCESS_U32) {
+    return ret.data[0];
+  } else {
+    return 0;
+  }
 }
 
 int led_on(int led_num) {
-  return command(DRIVER_NUM_LEDS, 1, led_num, 0);
+  syscall_return_t ret = command2(DRIVER_NUM_LEDS, 1, led_num, 0);
+  if (ret.type == TOCK_SYSCALL_SUCCESS) {
+    return TOCK_SUCCESS;
+  } else {
+    return ret.data[0];
```

Comment:
if the LED capsule mistakenly returned a `success_u32()`, rather than sucess, this would just return some arbitrary value rather than a valid failure return code. @phil-levis do you consider it acceptable for userspace drivers to assume the correct success / failure types rather than explicitly checking for them?

Reply:
Just a thought, but what about having a macro IS_SUCCESS that would test for all the SUCCESS cases?

Reply:
After looking at the alarm porting PR this seems like the expected approach, so marking as resolved

Reply:
Oh sorry didn't see your comment. The issue is that a given subcommand should only return a single success type, so I think you would want multiple macros (one expecting `TOCK_SYSCALL_SUCCESS`, one expecting `TOCK_SUCCESS_U32`, and so forth). But I am not sure if the ability to write slightly terser code is worth the added indirection

Reply:
> a macro IS_SUCCESS that would test for all the SUCCESS cases?

We explicitly don't want that. A command is only allowed to have a single success and failure variant, and as such all other success variants must be treated as an error, ideally a `BADRVAL` in userspace (as done in `libtock-rs`, as far as I know).